src/ui/theme/theme.ts

```diff
@@ -0,0 +1,31 @@
+import { BrandingAppearance } from "../../networking/responses/branding-response";
+import {
+  toFormColors,
+  toFormStyleVar,
+  toProductInfoStyleVar,
+  toShape,
+} from "./utils";
+import { Shape } from "./shapes";
+import { Colors } from "./colors";
+
+export class Theme {
+  constructor(
+    private readonly brandingAppearance?: BrandingAppearance | undefined,
+  ) {}
+
+  get shape(): Shape {
+    return toShape(this.brandingAppearance);
```

Comment:
Hmm this means these need to be recalculated every time they are called correct? Would it be possible to precalculate them on the constructor? That way we can just make `Theme` an interface.

Reply:
Yeah, not a bit calculation for now, but we can definitely calculate it when initializing.

In the end we are using a property, so changing from calculation to pre-calculated field will be transparent to the client. They won't need to change the usage `theme.shape`.

I don't feel it that urgent (calculations are really limited here), but I can change it if needed.

Reply:
No, I think it's not urgent, just an optimization IMO but can be addressed later